src/main/java/com/soboleoko/medicalclinic/PatientService.java

```diff
@@ -0,0 +1,41 @@
+package com.soboleoko.medicalclinic;
+
+import com.soboleoko.medicalclinic.model.Patient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PatientService {
+
+    private final PatientRepository patientRepository;
+
+    @Autowired
+    public PatientService(PatientRepository patientRepository) {
+        this.patientRepository = patientRepository;
+    }
+
+    public List<Patient> getAllPatients() {
+        return patientRepository.getAll();
+    }
+
+    public Optional<Patient> getPatientByEmail(String email) {
```

Comment:
Nie powinnismy do kontrolera zwracać Optionali

Reply:
Serwis powinien zwrocic do Kontrolera to co powinno znaleźć się w body

Reply:
Czyli tutaj powininsmy miec obsluge tego co ma sie stac w sytuacji jak Patient bedzie nullem ->
czyli tutaj powinnismy miec 
return patientRepository.findByEmail(email).orElseThrow(new IllegalArgumentException("Patient with given email does not exist."))